src/journal/journal-file.c

```diff
@@ -86,7 +88,82 @@
 /* The mmap context to use for the header we pick as one above the last defined typed */
 #define CONTEXT_HEADER _OBJECT_TYPE_MAX
 
-static int journal_file_set_online(JournalFile *f) {
+static int journal_file_set_offline_fd(int fd) {
+        typeof(((Header *)0)->state) state = STATE_OFFLINE;
+        int r;
+
+        assert_cc(sizeof(state) == 1);
+        assert(fd >= 0);
+
+        r = pwrite(fd, &state, sizeof(state), offsetof(Header, state));
+        if (r < 0)
+                return r;
+
+        r = fsync(fd);
+        if (r < 0)
+                return r;
+
+        return r;
+}
+
+
+static int journal_file_set_offline_internal(JournalFile *f) {
```

Comment:
yeah, if it is guaramteed that the f object stays around, and the mmap stays around then yeah, we don't need to bother with pwrite(). The only reason to do the pwrite() thing was really that we could fork of a thread with a dupped() copy of the fd, and then completely forget about it, as the thread could on its own sync the file, update the state and close the fd.

However, if the object stays around anyway (because we need the atomic var as communication channel), then there's no point in bothering with pwrite()